vnext/Microsoft.ReactNative.Cxx/JSValueReader-WinRT.h

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <CppWinRTIncludes.h>
+#include <UI.Xaml.Markup.h>
+#include <chrono>
+
+#include "JSValue.h"
+#include "JSValueTreeReader.h"
+#include "StructInfo.h"
+
+#include "winrt/Microsoft.ReactNative.h"
+
+#include <string>
+
+namespace winrt::Microsoft::ReactNative {
+inline BYTE GetAFromArgb(DWORD v) {
+  return ((BYTE)((v & 0xFF000000) >> 24));
+}
+inline BYTE GetRFromArgb(DWORD v) {
+  return ((BYTE)((v & 0x00FF0000) >> 16));
+}
+inline BYTE GetGFromArgb(DWORD v) {
+  return ((BYTE)((v & 0x0000FF00) >> 8));
+}
+inline BYTE GetBFromArgb(DWORD v) {
+  return ((BYTE)((v & 0x000000FF)));
+}
+
+inline ui::Color ColorFromArgb(const JSValue &value) {
+  auto argb = static_cast<DWORD>(value.AsInt64());
+  return winrt::ColorHelper::FromArgb(GetAFromArgb(argb), GetRFromArgb(argb), GetGFromArgb(argb), GetBFromArgb(argb));
+}
+
+inline ui::Color UndefinedColor() noexcept {
+  static auto c = ColorHelper::FromArgb(1, 0, 0, 0);
+  return c;
+}
+
+inline xaml::Media::SolidColorBrush UndefinedSolidColorBrush() noexcept {
+  static auto b = xaml::Media::SolidColorBrush(UndefinedColor());
+  return b;
+}
+
+inline void ReadValue(IJSValueReader const &reader, /*out*/ ui::Color &value) {
+  value = UndefinedColor();
```

Comment:
>value = UndefinedColor(); [](start = 2, length = 25)

Consider to put it inside of 'else' statement to avoid assigning the value twice.